tests/classes/test_function.py

```diff
@@ -86,9 +86,129 @@ def test_create_empty_copy(edgelist1):
     assert E2.shape == (8, 0)
     for node in E2.nodes:
         assert len(E1.nodes.memberships(node)) == 0
-    assert E2._hypergraph == {"name": "test", "timestamp": "Nov. 20"}
-    for n in H.nodes:
-        assert H.nodes[n]["name"] == attr_dict[n]["name"]
+    assert E2["name"] == "test"
+    assert E2["timestamp"] == "Nov. 20"
+    with pytest.raises(XGIError):
+        author = E2["author"]
+    for n in E2.nodes:
+        assert E2.nodes[n]["name"] == attr_dict[n]["name"]
+
+
+def test_set_node_attributes(edgelist1):
+    attr_dict1 = {
```

Comment:
add test for what happens if trying to set attr for a node not in the hypergraph?
attr_dict = {0 : {"name" : "Nobody"}}
From the function itself it looks like no error is raised, should we raise one? (Same for the set_edge_attr)

Reply:
I'm torn on this, but my inclination is that it's okay if it fails silently. Like if you had a giant list of attribute data some of which corresponded to nodes in the hypergraph and others which didn't (For instance the Project 90 network data set), you would potentially have to remove the attributes connected to non-existent nodes for it to run.

Reply:
Okay, fair enough.